fbw-a380x/src/systems/instruments/src/EWD/elements/WdMemos.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  ClockEvents,
+  ConsumerSubject,
+  DisplayComponent,
+  EventBus,
+  FSComponent,
+  Subject,
+  VNode,
+} from '@microsoft/msfs-sdk';
+import { FormattedFwcText } from 'instruments/src/EWD/elements/FormattedFwcText';
+import { EwdSimvars } from 'instruments/src/EWD/shared/EwdSimvarPublisher';
+import EWDMessages from '@instruments/common/EWDMessages';
+
+interface WdMemosProps {
+  bus: EventBus;
+}
+
+const padEWDCode = (code: number) => code.toString().padStart(9, '0');
+
+export class WdMemos extends DisplayComponent<WdMemosProps> {
+  private readonly sub = this.props.bus.getSubscriber<ClockEvents & EwdSimvars>();
+
+  private readonly memosLeftSvgRef = FSComponent.createRef<SVGGraphicsElement>();
+
+  private readonly memosRightSvgRef = FSComponent.createRef<SVGGraphicsElement>();
+
+  private readonly memosLeft = Array.from(Array(10), (_, idx) =>
+    ConsumerSubject.create(this.sub.on(`memo_left_${idx + 1}`).whenChanged(), 0),
+  );
+
+  private readonly memosRight = Array.from(Array(10), (_, idx) =>
+    ConsumerSubject.create(this.sub.on(`memo_right_${idx + 1}`).whenChanged(), 0),
+  );
+
+  private readonly memosLeftFormatString = Subject.create('');
+
+  private readonly memosRightFormatString = Subject.create('');
+
+  public onAfterRender(node: VNode): void {
+    super.onAfterRender(node);
+
+    this.sub
+      .on('realTime')
```

Comment:
Any reason to not use the subs from `memoLeft` and `memoRight`  to set the strings? Then we would avoid running the logic if nothing changed, or?

Reply:
True. Will push a better version with the next commit